src/cart/crt_group.c

```diff
@@ -1623,20 +1637,25 @@ crt_hdlr_uri_lookup(crt_rpc_t *rpc_req)
 	 * step 2, if rank:tag is not found, lookup rank:tag=0
 	 */
 	ul_out->ul_tag = 0;
-	crt_grp_lc_lookup(grp_priv_primary, crt_ctx->cc_idx,
-			  g_rank, 0, &cached_uri, NULL);
+	crt_grp_lc_lookup(grp_priv_primary, crt_ctx->cc_idx, g_rank, 0, &cached_uri, NULL);
 	ul_out->ul_uri = cached_uri;
 	if (ul_out->ul_uri == NULL)
 		D_GOTO(out, rc = -DER_OOG);
 
 out:
+	ul_out->ul_rc = rc;
+
+	if (rc != DER_SUCCESS)
+		D_WARN("uri lookup of (rank=%d:tag=%d) group=%s failed; rc=%d\n", ul_in->ul_rank,
+		       ul_in->ul_tag, grp_priv->gp_pub.cg_grpid, rc);
+
```

Comment:
[Not request changes] Since `rc` may be `-DER_OOG`, I think `D_INFO` or lower would be more appropriate. (Also, you probably don't want to mess with NLT and friends.)